ValueChangedGenerator/ValueChangedGenerator/ValueChangedGenerator.SourceGenerator/Generator.cs

```diff
@@ -0,0 +1,132 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using static Microsoft.CodeAnalysis.CSharp.SyntaxFactory;
+
+namespace ValueChangedGenerator
+{
+    public class Generator
+    {
+        public CompilationUnitSyntax GeneratePartialDeclaration(INamedTypeSymbol container, ClassDeclarationSyntax classDecl)
+        {
+            var strDecl = (StructDeclarationSyntax)classDecl.ChildNodes().First(x => x is StructDeclarationSyntax);
+
+            var def = new RecordDefinition(strDecl);
+            var generatedNodes = GetGeneratedNodes(def).ToArray();
+
+            var newClassDecl = container.GetContainingTypesAndThis()
+                .Select((type, i) => i == 0
+                    ? ClassDeclaration(type.Name).GetPartialTypeDelaration().AddMembers(generatedNodes)
+                    : ClassDeclaration(type.Name).GetPartialTypeDelaration())
+                .Aggregate((a, b) => b.AddMembers(a));
```

Comment:
I think there should be a better way to support nested partial classes, but I'll keep this as is for now.